### ISSUE TITLE
factorplot singular {row|col}_order produces wrong plots

### ISSUE THREAD

**Issue by tku137** [mwaskom/seaborn#1296]
Using factorplot with multiple values in a list to {row|col}_order produces the desired plot, even when providing only a subset of values, as expected. Putting a value in {row|col}_order that does not exist in the data produces an empty plot, as expected. However, using only one value in the {row|col}_order list produces the same false plot ignoring {row|col} all the time, even when providing an element that does not exist in the data.

Examplary behaviour:

```python
import matplotlib.pyplot as plt
import seaborn as sns

data = sns.load_dataset('titanic')
```

Using only two values of possible 4, the outcome is as expected:
```python
sns.factorplot(x='class', y='fare', hue='sex', row='embark_town', row_order=['Southampton', 'Cherbourg'], kind='box', data=data)
plt.show()
```
![1](https://user-images.githubusercontent.com/3052212/31277539-aa7783a2-aaa1-11e7-8569-2ea70a8912da.png)

Using a non-existent value along with others, the outcome is as expected:
```python
sns.factorplot(x='class', y='fare', hue='sex', row='embark_town', row_order=['Southampton', 'non-existent'], kind='box', data=data)
plt.show()
```
![3](https://user-images.githubusercontent.com/3052212/31277594-fd7e8884-aaa1-11e7-9532-d9003d1f6651.png)

However, using only one element, to get an overview or for whatever reason. 
**Note that this is NOT the plot for the corresponding data!**
```python
sns.factorplot(x='class', y='fare', hue='sex', row='embark_town', row_order=['Southampton'], kind='box', data=data)
plt.show()
```
![2](https://user-images.githubusercontent.com/3052212/31277629-2a3bb25c-aaa2-11e7-9309-47e3a728f5b1.png)

Even using only a non-existent element produces the same output, even though the selected data doesn't even exist:
```python
sns.factorplot(x='class', y='fare', hue='sex', row='embark_town', row_order=['non-existent'], kind='box', data=data)
plt.show()
```
![5](https://user-images.githubusercontent.com/3052212/31278076-ef53bbf6-aaa3-11e7-8c31-90a79730f55a.png)

The produced plot is ignoring the {row|col}_order selection and treats it as a factorplot without {row|col}:
```python
sns.factorplot(x='class', y='fare', hue='sex', kind='box', data=data)
plt.show()
```
![download](https://user-images.githubusercontent.com/3052212/31278438-44866f1e-aaa5-11e7-9e68-994b20bc72ac.png)

I know that using only one element in {row|col}_order might be somewhat unconventional and can easily be avoided by not using the row parameter at all and filter the dataframe by yourself, but nonetheless this usage produces plots that do not represent the actual data.

**Comment by mwaskom** [comment]
Indeed, the code does not anticipate having a single row/column level that is a subset (or null set) https://github.com/mwaskom/seaborn/blob/master/seaborn/axisgrid.py#L647

**Comment by mwaskom** [comment]
Oh I bet that's what causes this too https://github.com/mwaskom/seaborn/issues/890

**Comment by mwaskom** [comment]
Fixed in #1453; thanks for reporting.
